data-prepper-plugins/s3-sink/src/test/java/org/opensearch/dataprepper/plugins/codec/parquet/ParquetOutputCodecTest.java

```diff
@@ -567,15 +595,33 @@ private List<Map<String, Object>> createParquetRecordsList(final InputStream inp
                 final RecordReader<Group> recordReader = columnIO.getRecordReader(pages, new GroupRecordConverter(schema));
                 for (int row = 0; row < rows; row++) {
                     final Map<String, Object> eventData = new HashMap<>();
-                    int fieldIndex = 0;
                     final SimpleGroup simpleGroup = (SimpleGroup) recordReader.read();
+                    final GroupType groupType = simpleGroup.getType();
+
+
                     for (Type field : schema.getFields()) {
+                        Object value;
+                        int fieldIndex = groupType.getFieldIndex(field.getName());
                         try {
-                            eventData.put(field.getName(), simpleGroup.getValueToString(fieldIndex, 0));
-                        } catch (Exception parquetException) {
-                            LOG.error("Failed to parse Parquet", parquetException);
-                        }
-                        fieldIndex++;
+                            switch (field.getName()) {
+                                case "name": value = simpleGroup.getString(fieldIndex, 0);
+                                            break;
+                                case "age": value = simpleGroup.getInteger(fieldIndex, 0);
+                                            break;
+                                case "myLong": value = simpleGroup.getLong(fieldIndex, 0);
+                                            break;
+                                case "myFloat": value = simpleGroup.getFloat(fieldIndex, 0);
+                                            break;
+                                case "myDouble": value = simpleGroup.getDouble(fieldIndex, 0);
+                                            break;
+                                case "myArray": value = extractStringList(simpleGroup, "myArray");
+                                            break;
+                                case "nestedRecord": value = extractNestedGroup(simpleGroup, "nestedRecord");
+                                            break;
+                                default: throw new IllegalArgumentException("Unknown field");
+                            }
+                        } catch (RuntimeException e) {continue;}
```

Comment:
I think we should remove this try-catch. Just let the exception propagate up.

Relatedly, there is an exception below. Can you remove it?

```
} catch (Exception parquetException) {
            LOG.error("Failed to parse Parquet", parquetException);
```